pkg/controllers/subnet/subnetbinding_handler.go

```diff
@@ -0,0 +1,103 @@
+package subnet
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/workqueue"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/vmware-tanzu/nsx-operator/pkg/apis/vpc/v1alpha1"
+)
+
+func requeueSubnetBySubnetBindingCreate(ctx context.Context, c client.Client, objNew client.Object, q workqueue.TypedRateLimitingInterface[reconcile.Request]) {
+	newBM := objNew.(*v1alpha1.SubnetConnectionBindingMap)
+	enqueueSubnets(ctx, c, newBM, q)
+}
+
+func requeueSubnetBySubnetBindingUpdate(ctx context.Context, c client.Client, objOld client.Object, objNew client.Object, q workqueue.TypedRateLimitingInterface[reconcile.Request]) {
+	newBM := objNew.(*v1alpha1.SubnetConnectionBindingMap)
+	oldBM := objOld.(*v1alpha1.SubnetConnectionBindingMap)
+
+	if newBM.Spec.TargetSubnetName == oldBM.Spec.TargetSubnetName {
+		return
+	}
+
+	if newBM.Spec.TargetSubnetName != "" {
+		if err := enqueue(ctx, c, newBM.Namespace, newBM.Spec.TargetSubnetName, q); err != nil {
+			log.Error(err, "Failed to enqueue the new target Subnet after SubnetConnectionBindingMap updates", "Namespace", newBM.Namespace, "Subnet", newBM.Spec.TargetSubnetName, "SubnetConnectionBindingMap", newBM.Name)
+		}
+	}
+	if oldBM.Spec.TargetSubnetName != "" {
+		// Enqueue to ensure the finalizer can be removed from the old target Subnet if it is not used.
+		enqueue(ctx, c, oldBM.Namespace, oldBM.Spec.TargetSubnetName, q)
```

Comment:
we better also log error here if enqueue failed?

Reply:
The error logs are provided in function `enqueue`, see https://github.com/vmware-tanzu/nsx-operator/pull/901/files/52bb2d785f0c1dbdb23a74efe0fc5b5d9ec67ed7#diff-51fe24c011631b55a1ac7cef9d82fd5764d219eb2952bd6308ea5defbb199e85R53